tools/test/audio/comp_run.sh

```diff
@@ -28,6 +28,7 @@ FN_IN=input.raw
 FN_OUT=output.raw
 FN_TRACE:=trace.txt # This is default value if FN_TRACE is not set via -e option
 VALGRIND=yes
+XTRUN=
```

Comment:
> VALGRIND=yes

I realize this is not part of this PR but `yes` is a totally different command than `true`. See `man yes` and try running `yes`. So how did this ever work? Should it be `VALGRIND=true`?

Don't make the frequent confusion between `if true; then` which _runs_ the command `true` that returns a success exit code, versus `if [ "true" ]; then` which is the same as `if [ "false" ];` or `if [ "anything_not_empty" ]; then`. In the latter, `[` is the command that returns an exit code.

Because of this confusion and the risk of `$string` starting with `-` I recommend to NEVER use `[ "$string" ]` but always `[ -n "$string" ]` instead.

Reply:
Oops, yes it should be VALGRIND=true. It's help text print. 

Those examples are quite scary, wrong code can look OK.